File: src/c#/Creational/FactoryMethod/FactoryMethod2/PayrollOffice.cs
using System;
namespace FactoryMethod2
{
    public class PayrollOffice
    {
		public void PaySalary(Employee employee)
		{
			string name = employee.name;
			string department = employee.department;
			double salary = employee.CalculateSalary();

			Console.WriteLine("Paying " + salary + " to " + " " + name + " in " + department);
		}
	}
}

File: src/typescript/com/gokselkucuksahin/dp/creational-patterns/factory/employee1/payroll-office.ts
import {Employee} from "./employee";

export class PayrollOffice {
  public paySalary(employee: Employee): void {
    const name: string = employee.getName();
    const department: string = employee.getDepartment();
    const salary: number = employee.getSalary();
    console.log(`Paying ${salary} to ${name} in ${department}`);
  }
}